### ISSUE TITLE
Add method to extract level configuration from path

### ISSUE THREAD

**Issue by Vittly** [bem/bem-sdk#277]
In [webpack-bem-plugin](https://github.com/bem/webpack-bem-plugin/pull/1#discussion_r165514501) I want to get preset from path to specific file in project.

Firstly I thought that `levelSync` solves my problem. But It returns "nearest parent config".

Then I read [Readme](https://github.com/bem/bem-sdk/tree/master/packages/config#config-example). This part:

```js
"levels": [
    {
        "path": "path/to/level",
        "scheme": "nested" // < --- Look here. May be "preset" or "naming" should be there
    }
],
```

There is no `method` to get options for specific layer. My suggestions:
* add method to return level declaration by path to file
* add method to return level declaration by level (if [naming.file.parse](https://github.com/bem/bem-sdk/issues/276) return level from path)
* tell me that "preset" or "naming" is project level option and put it in the root of `.bemrc` + add fix to readme

**Comment by qfox** [comment]
cc @tadatuta We need your expertise

**Comment by tadatuta** [comment]
You are right, `naming` is project level option and can be in the root of `.bemrc`.
Added it to example in readme: https://github.com/bem/bem-sdk/pull/286

>  I thought that levelSync solves my problem. But It returns "nearest parent config".

Actually it's not parent config but the merge of all the declarations for the level from all the configs (almost like `Object.assign(levelDeclFromTopLevelConf, levelDeclFromNearestLevelConf)`).

There's possibility to work with raw data with [config](https://github.com/bem/bem-sdk/tree/master/packages/config#configs-1) method but for sure it's not convenient.

**Comment by Vittly** [comment]
> Actually it's not parent config but the merge of all the declarations for the level from all the configs

Yes but except fields defined in `levelConf`.
Lets write simple example:

```js
// .bemrc
{
  prop1: 'foo',
  levels: [{ layer: 'bar', prop2: 'baz' }]
}

// code
const BemConf = require('@bem/sdk.config');
const config = new BemConf();

config.getLevelSync('bar');
// returns { prop1: 'foo' } <---- where is my prop2 ?
```

Issue is about two things:
1. `naming` should be top level property in example. Please fix https://github.com/bem/bem-sdk/pull/286/files#r170572472
2. somewhere should be a way to get properties from level declaration. Is it?

**Comment by tadatuta** [comment]
the issue here is that `level()` and `levelSync()` still expect path to level as an argument not layer name (see: https://github.com/bem/bem-sdk/blob/master/packages/config/index.js#L265).

so it your example `config.levelSync('bar');` is equivalent to `config.levelSync('anythingElse');` and returns just common config. with path is works how expected (https://github.com/bem/bem-sdk/pull/286/files#r170595399).

what we need is to support both path (as fallback) and layer as arguments for `level` methods.

**Comment by Vittly** [comment]
Agree 👍  [Here](https://github.com/bem/bem-sdk/pull/284/files#diff-621cc23937ce7f99fed2fab23dbbbee0R38) is noted that `path` is deprecated also

**Comment by qfox** [comment]
> what we need is to support both path (as fallback) and layer as arguments for level methods.

Wait, what you mean about `layer as arguments`?
We still have levels with one or more layers. And they are still can have separate config.

**Comment by veged** [comment]
Реализовано в 7373dcb (PR #398).

```ts
import { bemConfig } from '@bem/sdk.config';

const cfg = bemConfig({ cwd: '/project' });

// async
const level = await cfg.levelByPath('/project/common.blocks/button/button.css');

// sync
const level = cfg.levelByPathSync('/project/common.blocks/button/button.css');
```

Семантика:
- Возвращает level config, чей `path` является наиболее **специфичным** (longest) prefix'ом запрошенного пути.
- Уважает directory boundaries — `/a/b/blocks` НЕ матчится с `/a/b/blocks-extra/...`.
- Относительные пути резолвятся через `path.resolve(options.cwd, input)`.
- Возвращает `undefined` если ни один level не подходит.

Реализовано поверх существующего `levelMap()` / `levelMapSync()` — без дополнительного I/O. Покрытие: 7 кейсов в `packages/config/src/index.test.ts`.
